src/codegeneration/InlineES6ModuleTransformer.js

```diff
@@ -139,6 +138,31 @@ export class InlineES6ModuleTransformer extends ModuleTransformer {
   }
 
   /**
+   * Handles importing modules that have `export * from './path/to/foo.js'` declarations.
+   * @returns {Array} statements
+   */
+  appendExportStatement(statements) {
+    let exportObject = createObjectLiteral(this.getExportProperties());
+    if (this.exportVisitor_.starExports.length) {
+      let starExports = this.exportVisitor_.starExports;
+      let starIdents = starExports.map((moduleSpecifier) => {
+        return createIdentifierExpression(
+          this.getTempVarNameForModuleSpecifier(moduleSpecifier));
+      });
+      statements.push(parseStatement `function exportStar(s, d) { for (let k in s) d[k] = s[k]; }`);
+      statements.push(parseStatement `let starModule = {}`);
```

Comment:
`ModuleTransformer` extends `TempVarTransformer` so you can use `getTempIdentifier()`.

But it is not clear why we need to generate a function here? Can't it be inlined?

Also, do we want to use for-in+[[Get]]+[[Put]] over getOwnPropertyNames + getOwnProperty + defineProperty? It is not clear to me.